### PR TITLE
fix: ci integration tests workflow

### PR DESCRIPTION
**Pull request by MCozhusheck**
Description
---
After merging HTTP wallet connecting wallet to base node has changed but it was not reflected in the [integration tests](https://github.com/tari-project/tari/pull/7215/files#diff-5ebed9b65bb3499b6104a043654ed48edb7ff2e7a60af6d0d090d184ab0e425eL200). This PR fixes it since there is no longer gRPC method `set_base_node` but instead wallets needs to receive base_node HTTP address.
This also changes integration tests workflow to exit early if first test fails and exit with non-zero exit code.

How Has This Been Tested?
---
Run integration tests locally and check CI if it's still showing green check mark for failed tests.

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

## Summary by CodeRabbit

* **New Features**
  * Tests now expose HTTP ports for base node and wallet processes; base node startup disables the splash screen by default.

* **Bug Fixes**
  * Fixed a minor logging syntax issue in wallet integration tests.

* **Chores**
  * Upgraded test framework dependency and reformatted dependency lists.
  * CI/integration workflows: improved logging, explicit error handling, safer shell options, extended binary test timeout, and expanded workflow triggers.

### CODE REVIEW

File: integration_tests/src/wallet_process.rs [hunk 1]
```diff
@@ -162,7 +169,11 @@ pub async fn spawn_wallet(
 
         wallet_app_config.wallet.set_base_path(temp_dir_path.clone());
 
-        let rt = runtime::Builder::new_multi_thread().enable_all().build().unwrap();
+        let rt = runtime::Builder::new_multi_thread()
+            .thread_stack_size(4 * 1024 * 1024)// 4 MB stack size per thread (4 * 1024 * 1024 = 4,194,304 bytes)
```
Comment (MCozhusheck) [inline integration_tests/src/wallet_process.rs:174]: The default is 2 MB. In my cases running simple test case of sending one sided transaction resulted in thread panic with stack overflow. Increasing this number has resolved it.

**Comment by hansieodendaal** [comment]
l see many cucumber failures?

**Comment by MCozhusheck** [comment]
> l see many cucumber failures?

In this PR I planned only to fix integration tests workflow and address fixes for failing tests in next PR

File: integration_tests/src/wallet_process.rs [hunk 2]
```diff
@@ -143,6 +149,7 @@ pub async fn spawn_wallet(
         wallet_app_config.wallet.p2p.transport.transport_type = TransportType::Tcp;
         wallet_app_config.wallet.p2p.transport.tcp.listener_address =
             Multiaddr::from_str(&format!("/ip4/127.0.0.1/tcp/{}", port)).unwrap();
+        wallet_app_config.wallet.http_server_url = format!("http://127.0.0.1:{}", http_port);
```
Comment (hansieodendaal) [inline integration_tests/src/wallet_process.rs:152]: I run it locally like this to be sure it connects locally, otherwise I think it goes to the default fallback server
```suggestion
        wallet_app_config.wallet.http_server_url = format!("http://127.0.0.1:{}", http_port);
        wallet_app_config.wallet.fallback_http_server_url = format!("http://127.0.0.1:{}", http_port);
```

File: integration_tests/src/base_node_process.rs [hunk 3]
```diff
@@ -169,6 +174,7 @@ pub async fn spawn_base_node_with_config(
         base_node_config.base_node.grpc_address = Some(format!("/ip4/127.0.0.1/tcp/{}", grpc_port).parse().unwrap());
         base_node_config.base_node.report_grpc_error = true;
         base_node_config.base_node.metadata_auto_ping_interval = Duration::from_secs(3);
+        base_node_config.base_node.http_wallet_query_service.port = http_port.try_into().unwrap();
```
Comment (hansieodendaal) [inline integration_tests/src/base_node_process.rs:177]:
```suggestion
        base_node_config.base_node.http_wallet_query_service.port = http_port.try_into().unwrap();
        base_node_config.base_node.http_wallet_query_service.external_address = Some(format!("http://127.0.0.1::{}", http_port).parse().unwrap());
```

**Review by hansieodendaal — COMMENTED** [review]
If the purpose of the PR was to add in the HTTP detail it looks fine.

**Review by SWvheerden — COMMENTED** [review]
Looks good, just update those ports to use the fallback as well.

**Review by SWvheerden — DISMISSED** [review]

File: comms/core/src/protocol/messaging/error.rs [hunk 4]
```diff
@@ -34,6 +34,7 @@ use crate::{
 };
 
 /// Error type for inbound messages.
+#[allow(dead_code)]
```
Comment (SWvheerden) [inline comms/core/src/protocol/messaging/error.rs]: rather delete

File: comms/core/src/transports/hidden_service_transport.rs [hunk 5]
```diff
@@ -33,6 +33,7 @@ use crate::{
 
 const LOG_TARGET: &str = "comms::transports::hidden_service_transport";
 
+#[allow(dead_code)]
```
Comment (SWvheerden) [inline comms/core/src/transports/hidden_service_transport.rs]: delete

File: comms/dht/src/rpc/mock.rs [hunk 6]
```diff
@@ -32,6 +32,7 @@ use crate::{
     rpc::DhtRpcService,
 };
 
+#[allow(dead_code)]
```
Comment (SWvheerden) [inline comms/dht/src/rpc/mock.rs]: delete

**Review by SWvheerden — COMMENTED** [review]
looks good, please rather delete the unused structs than mark them as dead code
